batch/test/test_dag.py

```diff
@@ -304,3 +304,63 @@ def test_no_parents_allowed_without_batches(client):
         assert re.search('.*invalid parent batch: .*', err.response.text)
         return
     assert False
+
+
+def test_output_files_no_service_account_is_error(client):
+    batch = client.create_batch()
+    try:
+        batch.create_job('alpine:3.8',
+                         command=['/bin/sh', '-c', 'echo head > /out'],
+                         output_files=[('/out', 'gs://hail-ci-0-1-batch-volume-test-bucket')])
+    except requests.exceptions.HTTPError as err:
+        assert err.response.status_code == 400
+        assert re.search('.*invalid request: service account may be specified '
+                         'if and only if input_files or output_files is set.*',
+                         err.response.text)
+        return
+    assert False
+
+
+def test_input_files_no_service_account_is_error(client):
+    batch = client.create_batch()
+    try:
+        batch.create_job('alpine:3.8',
+                         command=['/bin/sh', '-c', 'echo head > /out'],
+                         input_files=[('gs://hail-ci-0-1-batch-volume-test-bucket', '/in')])
+    except requests.exceptions.HTTPError as err:
+        assert err.response.status_code == 400
+        assert re.search('.*invalid request: service account may be specified '
+                         'if and only if input_files or output_files is set.*',
+                         err.response.text)
+        return
+    assert False
+
+
+def test_service_account_no_files_is_error(client):
+    batch = client.create_batch()
+    try:
+        batch.create_job('alpine:3.8',
+                         command=['/bin/sh', '-c', 'echo head > /out'],
+                         copy_service_account_name='batch-volume-tester')
+    except requests.exceptions.HTTPError as err:
+        assert err.response.status_code == 400
+        assert re.search('.*invalid request: service account may be specified '
```

Comment:
Why do we have such a strict rule on this? Why is it bad if I specify the service account but no other files?

Reply:
If no files are specified, the copy tasks are not created and the service account is unused, so that seems like a bug.